cli/domains_test.go

```diff
@@ -0,0 +1,728 @@
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/absmach/magistrala/cli"
+	"github.com/absmach/magistrala/internal/testsutil"
+	"github.com/absmach/magistrala/pkg/errors"
+	svcerr "github.com/absmach/magistrala/pkg/errors/service"
+	mgsdk "github.com/absmach/magistrala/pkg/sdk/go"
+	sdkmocks "github.com/absmach/magistrala/pkg/sdk/mocks"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+const (
+	createCommand     = "create"
+	getCommand        = "get"
+	updateCommand     = "update"
+	deleteCommand     = "delete"
+	usersCommand      = "users"
+	enableCommand     = "enable"
+	disableCommand    = "disable"
+	connectCommand    = "connect"
+	disconnectCommand = "disconnect"
+	connectionsCmd    = "connections"
+	assignCommand     = "assign"
+	unassignCommand   = "unassign"
+	shareCmd          = "share"
+	unshareCmd        = "unshare"
+)
+
+var domain = mgsdk.Domain{
+	ID:    testsutil.GenerateUUID(&testing.T{}),
+	Name:  "testdomain",
+	Alias: "alias",
+}
+
+func TestCreateDomainsCmd(t *testing.T) {
+	sdkMock := new(sdkmocks.SDK)
+	cli.SetSDK(sdkMock)
+	domainCmd := cli.NewDomainsCmd()
+	rootCmd := setFlags(domainCmd)
+
+	var dom mgsdk.Domain
+
+	cases := []struct {
+		desc          string
+		args          []string
+		domain        mgsdk.Domain
+		errLogMessage string
+		sdkErr        errors.SDKError
+		logType       outputLog
+	}{
+		{
+			desc: "create domain successfully",
+			args: []string{
+				createCommand,
+				dom.Name,
+				dom.Alias,
+				validToken,
+			},
+			logType: entityLog,
+			domain:  domain,
+		},
+		{
+			desc: "create domain with invalid args",
+			args: []string{
+				createCommand,
+				dom.Name,
+				dom.Alias,
+				validToken,
+				extraArg,
+			},
+			logType: usageLog,
+		},
+		{
+			desc: "create domain with invalid token",
+			args: []string{
+				createCommand,
+				dom.Name,
+				dom.Alias,
+				invalidToken,
+			},
+			sdkErr:        errors.NewSDKErrorWithStatus(svcerr.ErrAuthorization, http.StatusUnauthorized),
+			errLogMessage: fmt.Sprintf("\nerror: %s\n\n", errors.NewSDKErrorWithStatus(svcerr.ErrAuthorization, http.StatusUnauthorized)),
+			logType:       errLog,
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.desc, func(t *testing.T) {
+			sdkCall := sdkMock.On("CreateDomain", mock.Anything, mock.Anything).Return(tc.domain, tc.sdkErr)
+			out := executeCommand(t, rootCmd, tc.args...)
+
+			switch tc.logType {
+			case entityLog:
+				err := json.Unmarshal([]byte(out), &dom)
+				assert.Nil(t, err)
+				assert.Equal(t, tc.domain, dom, fmt.Sprintf("%s unexpected response: expected: %v, got: %v", tc.desc, tc.domain, dom))
+			case errLog:
+				assert.Equal(t, tc.errLogMessage, out, fmt.Sprintf("%s unexpected error response: expected %s got errLogMessage:%s", tc.desc, tc.errLogMessage, out))
+			case usageLog:
+				assert.False(t, strings.Contains(out, rootCmd.Use), fmt.Sprintf("%s invalid usage: %s", tc.desc, out))
+			}
+			sdkCall.Unset()
+		})
```

Comment:
```suggestion
		for _, tc := range cases {
			sdkCall := sdkMock.On("CreateDomain", mock.Anything, mock.Anything).Return(tc.domain, tc.sdkErr)
			out := executeCommand(t, rootCmd, tc.args...)

			switch tc.logType {
			case entityLog:
				err := json.Unmarshal([]byte(out), &dom)
				assert.Nil(t, err)
				assert.Equal(t, tc.domain, dom, fmt.Sprintf("%s unexpected response: expected: %v, got: %v", tc.desc, tc.domain, dom))
			case errLog:
				assert.Equal(t, tc.errLogMessage, out, fmt.Sprintf("%s unexpected error response: expected %s got errLogMessage:%s", tc.desc, tc.errLogMessage, out))
			case usageLog:
				assert.False(t, strings.Contains(out, rootCmd.Use), fmt.Sprintf("%s invalid usage: %s", tc.desc, out))
			}
			sdkCall.Unset()
	}
```

same for all other cases